qt/Torrent.h

```diff
@@ -118,12 +119,12 @@ class TorrentHash
 
     explicit TorrentHash(char const* str)
     {
-        tr_hex_to_sha1(data_.data(), str);
+        data_ = tr_sha1_from_string(str != nullptr ? str : "");
     }
 
     explicit TorrentHash(QString const& str)
     {
-        tr_hex_to_sha1(data_.data(), str.toUtf8().constData());
+        data_ = tr_sha1_from_string(str.toUtf8().constData());
```

Comment:
Since we're allocating memory (for `QByteArray`) here anyway, using `std::string` (which casts to `std::string_view` implicitly) would probably be more efficient (than having to calculate length from a char pointer).
```suggestion
        data_ = tr_sha1_from_string(str.toStdString());
```

Reply:
Good catch. Changed in 8ae6d030a6dfd753c18d644af83e626269b883fd